src/twister2/device/native_simulator_adapter.py

```diff
@@ -23,78 +25,86 @@ def __init__(self, twister_config, hardware_map: HardwareMap | None = None, **kw
         self._process: subprocess.Popen | None = None
         self._process_ended_with_timeout: bool = False
         self.queue: Queue = Queue()
+        self._stop_job: bool = False
 
     @staticmethod
-    def get_command(build_dir: Path | str) -> str:
-        return str((Path(build_dir) / 'zephyr' / 'zephyr.exe').resolve())
+    def _get_command(build_dir: Path | str) -> list[str]:
+        """
+        Return command to run.
+
+        :param build_dir: build directory
+        :return: command to run
+        """
+        return [str((Path(build_dir) / 'zephyr' / 'zephyr.exe').resolve())]
 
     def connect(self, timeout: float = 60):
         pass
 
-    def disconnect(self):
-        """End subprocess."""
-        if self._process:
-            if self._process.poll() is None:
-                self._process.kill()
-                logger.debug('Process terminated: %s', self._process.pid)
-                self._process = None
+    async def _run_command(self, command: list[str], timeout: float = 60):
+        assert isinstance(command, (list, tuple, set))  # to avoid stupid and  difficult to debug mistakes
+        self._process = await asyncio.create_subprocess_exec(
+            *command,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.STDOUT,
+            env=self.env
+        )
+        logger.debug('Started subprocess with PID %s', self._process.pid)
+        end_time = time.time() + timeout
+        while not self._stop_job:
+            try:
+                line = await asyncio.wait_for(self._process.stdout.readline(), 0.1)
+            except asyncio.TimeoutError:
+                pass
+            else:
+                if line:
+                    self.queue.put(line.decode('utf-8').strip())
+                    logger.debug(line.decode('utf-8').strip())
```

Comment:
This logger is probably unnecessary - the same logs are printed in `ztest_log_parser`.

Reply:
Removed